solr/core/src/java/org/apache/solr/handler/admin/ColStatus.java

```diff
@@ -201,32 +204,35 @@ public void getColStatus(NamedList<Object> results) {
             }
             QueryRequest req = new QueryRequest(params);
             NamedList<Object> rsp = client.request(req);
-            rsp.remove("responseHeader");
-            leaderMap.add("segInfos", rsp);
-            NamedList<?> segs = (NamedList<?>) rsp.get("segments");
+            final var segmentResponse =
+                SolrJacksonMapper.getObjectMapper().convertValue(rsp, GetSegmentDataResponse.class);
+            segmentResponse.responseHeader = null;
+
+            final var segs = segmentResponse.segments;
             if (segs != null) {
-              for (Map.Entry<String, ?> entry : segs) {
-                NamedList<Object> fields =
-                    (NamedList<Object>) ((NamedList<Object>) entry.getValue()).get("fields");
-                if (fields != null) {
-                  for (Map.Entry<String, Object> fEntry : fields) {
-                    Object nc = ((NamedList<Object>) fEntry.getValue()).get("nonCompliant");
-                    if (nc != null) {
+              for (Map.Entry<String, GetSegmentDataResponse.SingleSegmentData> entry :
+                  segs.entrySet()) {
+                final var fieldInfoByName = entry.getValue().fields;
```

Comment:
I love how typed this is now.  This is an important part of the V2 vision coming into reality.